homeassistant/components/google_assistant/trait.py

```diff
@@ -362,23 +363,26 @@ class ColorSettingTrait(_Trait):
     commands = [COMMAND_COLOR_ABSOLUTE]
 
     @staticmethod
-    def supported(domain, features, device_class):
+    def supported(domain, features, device_class, attributes):
         """Test if state is supported."""
         if domain != light.DOMAIN:
             return False
 
-        return features & light.SUPPORT_COLOR_TEMP or features & light.SUPPORT_COLOR
+        color_modes = attributes.get(light.ATTR_SUPPORTED_COLOR_MODES, [])
+        return light.COLOR_MODE_COLOR_TEMP in color_modes or any(
+            mode in color_modes for mode in light.COLOR_MODES_COLOR
+        )
 
     def sync_attributes(self):
         """Return color temperature attributes for a sync request."""
         attrs = self.state.attributes
-        features = attrs.get(ATTR_SUPPORTED_FEATURES, 0)
+        color_modes = attrs.get(light.ATTR_SUPPORTED_COLOR_MODES, [])
         response = {}
 
-        if features & light.SUPPORT_COLOR:
+        if any(mode in color_modes for mode in light.COLOR_MODES_COLOR):
```

Comment:
Don't default `color_modes` to an empty list and check if `color_modes` is not None. Also turn this check around, `color_modes` is shorter.